pages/bookmarks.jsx

```diff
@@ -45,50 +48,113 @@ export default React.createClass({
+        });
+      }
+    });
+
+    // and clear ls bookmarks
   },
-  getloggedInStatusContent() {
-    if (user.loggedin) {
-      return this.getContentForLoggedInUser();
+  getContentForLoggedInUser() {
+    // console.log(`> getContentForLoggedInUser`);
+    let importHint;
+
+    if (this.state.lsBookmarkedIds.length > 0) {
+      importHint = <span>Did you add favs while logged out? You can <button className="btn btn-link inline-link" onClick={evt=>this.handleImportBookmarksClick(evt)}>import favs</button> from this browswer.</span>;
     }
 
-    if (user.failedLogin) {
-      return this.getFailurePrompt();
+    if (this.state.bookmarksImported) {
+      importHint = <span>Your favs have been imported.</span>;
     }
 
-    return this.getAnonymousContent();
+    if (importHint) {
+      importHint = <p><small className="text-muted">{importHint}</small></p>;
+    }
+
+    return <div>
+            <p className={classNames({'mb-0': !!importHint})}>Hi {user.username}! <button onClick={this.handleLogOutBtnClick} className="btn btn-link inline-link">Log out</button>.</p>
+            {importHint}
+            <ProjectLoader bookmarkedOnly={true} />
+          </div>;
   },
-  getBookmarkedProjects() {
-    // We have renamed all non user facing "favorites" related variables and text (e.g., favs, faved, etc) to "bookmarks".
-    // This is because we want client side code to match what Pulse API uses (i.e., bookmarks)
-    // For user facing bits like UI labels and URL path we want them to stay as "favorites".
-    // For more info see: https://github.com/mozilla/network-pulse/issues/326
-    let headerText = `Save your Favs`;
-    let bookmarkedProjects = (<HintMessage iconComponent={<img src="/assets/svg/icon-bookmark-selected.svg" />}
-                                           header={headerText}
-                                           linkComponent={<Link to={`/featured`}>Explore featured</Link>}>
-                                <p>Tap the heart on any project to save it here.</p>
-                              </HintMessage>);
+  getAnonymousContent() {
+    // console.log(`> getAnonymousContent`);
+    let signInPrompt = <p><button className="btn btn-link inline-link" onClick={this.handleSignInBtnClick}>Sign in with Google</button>.</p>;
+    let bookmarkedProjects = this.renderHintMessage();
+
+    if (user.failedLogin) {
+      signInPrompt = <p>Sorry, login failed! Please try again or <a href="mailto:https://mzl.la/pulse-contact">contact us</a>.</p>;
+    }
 
-    if (this.state.bookmarkedIds.length > 0) {
-      bookmarkedProjects = <ProjectLoader ids={this.state.bookmarkedIds} />;
+    if (this.state.lsBookmarkedIds.length > 0) {
+      bookmarkedProjects = <ProjectLoader ids={this.state.lsBookmarkedIds} />;
     }
 
-    return bookmarkedProjects;
+    return <div>
+            {signInPrompt}
+            {bookmarkedProjects}
+          </div>;
+  },
+  renderContent() {
+    if (user.loggedin === undefined) return null;
```

Comment:
It takes a few seconds to find out if the current user is logged in or not. We shouldn't assume the current user is anonymous and show him/her "Please sign in" message. I changed the default value for `user.loggedin` to `null` (instead of `false`) so we can show a blank screen while getting result back from the `/userstatus` call.